src/plugins/visualize/public/application/utils/get_table_columns.tsx

```diff
@@ -108,13 +108,7 @@ export const getTableColumns = (
       // In case an error occurs i.e. the vis has wrong type, we render the vis but without the link
       !error ? (
         <EuiLink
-          onClick={() => {
-            if (editApp) {
-              application.navigateToApp(editApp, { path: editUrl });
-            } else if (editUrl) {
-              history.push(editUrl);
-            }
-          }}
+          href={editApp ? application.getUrlForApp(editApp, { path: editUrl }) : `#${editUrl}`}
```

Comment:
@BSFishy I really like the direction you've gone with this, but unfortunately it's not sufficient to hardcode the `#` prefix to the `editUrl`. The reason that works for the `wizard` plugin is that we define `#/` as the `aliasPath` here: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/wizard/public/plugin.ts#L117

My suspicion is that we may need to adjust the way `editUrl` is fetched for aliased visualizations upstream of this component so that it correctly accounts for the `aliasPath`.

Reply:
I just did a bit of digging, and I think this should work fine. The reason that it works with the `wizard` plugin is because it defines its own application, causing it to call `getUrlForApp`. Otherwise, it will follow the same functionality that was done by `navigateToUrl`. In that function, a call was made to `history.push`, which is implemented in the `history` package. In that package, they simply prepend a `#` to the beginning of the path:

https://github.com/remix-run/history/blob/3e9dab413f4eda8d6bce565388c5ddb7aeff9f7e/packages/history/index.ts#L693

I can still have it get the url upstream of this component, though, like the `DashboardListing` component does, however in that component, the `getViewUrl` function is a property, which I think would be a breaking change.

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/48fe60b40661c0bb19ed73d3426f5d5ec418887c/src/plugins/dashboard/public/application/listing/dashboard_listing.js#L173-L177

Reply:
Ah, thanks for digging into the `history` implementation, that makes sense.